cmake/CMakeshift/modules/FindHSL.cmake

```diff
@@ -0,0 +1,56 @@
+
+#.rst:
+# FindHSL
+# --------
+#
+# Find the HSL library.
+#
+# This will define the following variables::
+#
+#   HSL_FOUND      - True if the HSL library was found
+#
+# and the following imported targets::
+#
+#   HSL::mc64       - http://www.hsl.rl.ac.uk/catalogue/mc64.html
+#
+
+include(FindPackageHandleStandardArgs)
+
+set(HSL_FOUND FALSE)
+
+if (NOT HSL_FIND_COMPONENTS)
+    set(HSL_FIND_COMPONENTS mc64 mc32)
+    foreach (_hsl_component ${HSL_FIND_COMPONENTS})
```

Comment:
Nitpick: in modern CMake prefer `foreach(_hsl_component IN LISTS HSL_FIND_COMPONENTS)` for readability.